File: api/routes/auth.route.go
package routes

import (
	"github.com/gin-gonic/gin"
	"api/controllers"
)

func AuthRoutes(router *gin.Engine) {
	auth := router.Group("/api/auth")
	{
		auth.POST("/signup", controllers.Signup)
		// You can add other auth routes here:
		 auth.POST("/signin", controllers.Signin)
		 auth.POST("/forgot-password", controllers.ForgotPassword)
		 auth.POST("/reset-password", controllers.ResetPassword)

		 auth.POST("/adminRegister", controllers.AdminRegister)
		auth.POST("/adminLogin", controllers.AdminLogin)
	}
}

File: api/routes/auth.route.js
import express from "express";
import {
  forgotPassword,
  google,
  loginAdmin,
  resetPassword,
  signOut,
  signin,
  signup,
} from "../controllers/auth.controller.js";

const router = express.Router();

router.post("/adminLogin", loginAdmin);
router.post("/signup", signup);
router.post("/signin", signin);
router.post("/google", google);
router.get("/signout", signOut);
router.post("/forgot-password", forgotPassword);
router.post("/reset-password", resetPassword);

export default router;
